modules/mo_pytorch/mo_extensions/load/pytorch/hooks.py

```diff
@@ -138,10 +134,27 @@ def size(self, dim=None):
         return self.shape if dim is None else self.dynamic_shape[dim]
 
     def clone(self):
-        return self
+        class Identity(nn.Module):
+            def __init__(self):
+                super().__init__()
+
+        return forward_hook(Identity(), (self,))
 
     def to(self, dtype):
-        return self
+        mapping = {
+            torch.int32: np.int32,
+            torch.int64: np.int64,
+            torch.float32: np.float32,
+        }
+        if not dtype in mapping:
+            raise Exception("Unexpected dst type:", dtype)
+
+        class Cast(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.dst_type = mapping[dtype]
```

Comment:
Don't we need to pass `dtype` as a constructor parameter?

Reply:
Yes, but actually class uses a global `dtype` from method's parameters. Probably that's not clear and better to refactor by a separate PR. Local class definition is also a problem but not a critical one.